ci: add automated wiki buffer documentation workflow

alandtse opened the pull request:
- Create reusable prepare-shaders composite action to eliminate duplication
- Add update-buffers-wiki workflow to automatically update wiki on dev pushes
- Refactor shader-validation job to use prepare-shaders action
- Use Andrew-Chen-Wang/github-wiki-action for wiki updates
- Run hlslkit-buffer-scan to generate buffer register usage documentation

The new workflow triggers on shader changes to dev branch and updates https://github.com/doodlum/skyrim-community-shaders/wiki/Buffers with a comprehensive table of all buffer register assignments, types, and feature usage.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Automatic shader buffer documentation is generated and synced to the project wiki on relevant code updates.
  * Added a reusable "Prepare Shaders" CI action for shader preparation.

* **Chores**
  * Centralized shader preparation in CI, simplifying workflows and removing legacy multi-step setup.
  * Simplified shader validation flow and improved CI reliability and caching.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alandtse committed:
chore: address ai comments

alandtse committed:
style: pre-commit

davo0411 reviewed (approved):
looks good, was sorely needed.

if tested and working i am happy to merge

alandtse committed:
ci: add automated wiki buffer documentation workflow

- Create reusable prepare-shaders composite action to eliminate duplication
- Add update-buffers-wiki workflow to automatically update wiki on dev pushes
- Refactor shader-validation job to use prepare-shaders action
- Use Andrew-Chen-Wang/github-wiki-action for wiki updates
- Run hlslkit-buffer-scan to generate buffer register usage documentation

The new workflow triggers on shader changes to dev branch and updates
https://github.com/doodlum/skyrim-community-shaders/wiki/Buffers with
a comprehensive table of all buffer register assignments, types, and
feature usage.

Co-authored-by: factory-droid[bot] <138933559+factory-droid[bot]@users.noreply.github.com>

# Conflicts:
#	.github/workflows/build.yaml

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# interactive rebase in progress; onto 8b8b375d
# Last command done (1 command done):
#    pick a379eed6 # ci: add automated wiki buffer documentation workflow
# Next commands to do (2 remaining commands):
#    pick 7b6f4f60 # chore: address ai comments
#    pick b4311dfb # style: pre-commit
# You are currently rebasing branch 'buffer_scan_ci' on '8b8b375d'.
#
# Changes to be committed:
#	new file:   .github/actions/prepare-shaders/action.yml
#	modified:   .github/workflows/build.yaml
#	new file:   .github/workflows/update-buffers-wiki.yaml
#
# Changes not staged for commit:
#	modified:   extern/CommonLibSSE-NG (new commits, untracked content)
#
# Untracked files:
#	AUTO_GENERATED_COMPATIBILITY.md
#	extern/NVAPI/
#	tools/feature_compatibility_matrix.sh
#
